src/Raven.Server/Documents/Indexes/IndexContext.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Diagnostics;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Indexes;
+using Raven.Server.Config.Categories;
+using Raven.Server.Documents.Indexes.Auto;
+using Raven.Server.Documents.Indexes.Errors;
+using Raven.Server.Documents.Indexes.MapReduce.Auto;
+using Raven.Server.Documents.Indexes.MapReduce.Static;
+using Raven.Server.Documents.Indexes.Static;
+using Raven.Server.Documents.Indexes.Static.Counters;
+using Raven.Server.Documents.Indexes.Static.TimeSeries;
+
+namespace Raven.Server.Documents.Indexes;
+
+public class StaticIndexContext : IndexContext
+{
+    public StaticIndexContext([NotNull] IndexDefinitionBaseServerSide definition, [NotNull] IndexingConfiguration configuration, AbstractStaticIndexBase staticIndex)
+        : base(definition, configuration)
+    {
+        Compiled = staticIndex;
+    }
+
+    public StaticIndexContext([NotNull] Index index)
+        : base(index)
+    {
+        Compiled = index switch
+        {
+            MapIndex mapIndex => mapIndex._compiled,
+            MapReduceIndex mapReduceIndex => mapReduceIndex._compiled,
+            MapTimeSeriesIndex mapTimeSeries => mapTimeSeries._compiled,
+            MapCountersIndex mapCountersIndex => mapCountersIndex._compiled,
+            _ => throw new ArgumentOutOfRangeException(nameof(index))
+        };
+    }
+
+    public readonly AbstractStaticIndexBase Compiled;
+}
+
+public class IndexContext
```

Comment:
My proposals for the name to avoid having one more "context"

- `IndexBasicInfo` 
- `IndexInfoContainer` 
- `IndexInfoHolder`
- `IndexComponents`
- `IndexIngredients`